pkg/controllers/job/helpers/helpers_test.go

```diff
@@ -35,23 +35,33 @@ func TestCompareTask(t *testing.T) {
 		expect bool
 	}{
 		{
-			generateTaskInfo("prod-worker-21", createTime),
-			generateTaskInfo("prod-worker-1", createTime),
+			generateTaskInfo("prod-worker-21", createTime, "pod-1"),
+			generateTaskInfo("prod-worker-1", createTime, "pod-2"),
 			false,
 		},
 		{
-			generateTaskInfo("prod-worker-0", createTime),
-			generateTaskInfo("prod-worker-3", createTime),
+			generateTaskInfo("prod-worker-0", createTime, "pod-1"),
+			generateTaskInfo("prod-worker-3", createTime, "pod-2"),
 			true,
 		},
 		{
-			generateTaskInfo("prod-worker", createTime),
-			generateTaskInfo("prod-worker-3", createTime.Add(time.Hour)),
+			generateTaskInfo("prod-worker", createTime, "pod-1"),
+			generateTaskInfo("prod-worker-3", createTime.Add(time.Hour), "pod-2"),
 			true,
 		},
 		{
-			generateTaskInfo("prod-worker", createTime),
-			generateTaskInfo("prod-worker-3", createTime.Add(-time.Hour)),
+			generateTaskInfo("prod-worker", createTime, "pod-1"),
```

Comment:
We'd better add two test case that task index parsed failed, and one case is two pods with same timestamp, another case is two pods with differenet timestamp.

Reply:
case 3&4: task index parsed failed but two pods with different timestamp
case 5&6: task index parsed failed and two pods with same timestamp

Reply:
@Monokaix hi, is there anything else that needs to be modified?

Reply:
Ok, will trigger the CI later: )